src/code/audio_seqplayer.c

```diff
@@ -1,74 +1,1077 @@
+    return instId;
+}
+
+void Audio_SetInstrument(SequenceChannel* seqChannel, u8 instId) {
+    if (instId >= 0x80) {
+        seqChannel->instOrWave = instId;
+        seqChannel->instrument = NULL;
+    } else if (instId == 0x7F) {
+        seqChannel->instOrWave = 0;
+        seqChannel->instrument = (Instrument *) 1;
+    } else if (instId == 0x7E) {
+        seqChannel->instOrWave = 1;
+        seqChannel->instrument = (Instrument *) 2;
+    } else if ((seqChannel->instOrWave =
+            Audio_GetInstrument(seqChannel, instId, &seqChannel->instrument, &seqChannel->adsr)) == 0) {
+        seqChannel->hasInstrument = 0;
+        return;
+    }
+    seqChannel->hasInstrument = 1;
+}
+
+void Audio_SequenceChannelSetVolume(SequenceChannel* seqChannel, u8 volume) {
+    seqChannel->volume = (f32) (s32) volume / 127.0f;
+}
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/audio_seqplayer/Audio_SequenceChannelProcessScript.s")
+
+void Audio_SequenceChannelProcessScript(SequenceChannel* seqChannel);
+
+#pragma GLOBAL_ASM("asm/non_matchings/code/audio_seqplayer/Audio_SequencePlayerProcessSequence.s")
+
+void Audio_SequencePlayerProcessSequence(SequencePlayer* seqPlayer);
+
+void Audio_ProcessSequences(s32 arg0) {
+    SequencePlayer *seqPlayer;
+    u32 i;
+
+    gAudioContext.gNoteSubEuOffset = (gAudioContext.gAudioBufferParameters.unk_08 - arg0 - 1) * gAudioContext.gMaxSimultaneousNotes;
+    for (i = 0; i < (u32) gAudioContext.gAudioBufferParameters.unk_10; i++) {
+        seqPlayer = &gAudioContext.gSequencePlayers[i];
+        if (seqPlayer->enabled == 1) {
+            Audio_SequencePlayerProcessSequence(seqPlayer);
+            Audio_SequencePlayerProcessSound(seqPlayer);
+        }
+    }
+    Audio_ProcessNotes();
+}
+
+void Audio_ProcessSequence(SequencePlayer *seqPlayer) {
+    while (seqPlayer->unk_DC > 0) {
+        Audio_SequencePlayerProcessSequence(seqPlayer);
+        Audio_SequencePlayerProcessSound(seqPlayer);
+        seqPlayer->unk_DC--;
+    }
+}
+
+void Audio_ResetSequencePlayer(SequencePlayer *seqPlayer) {
+    s32 i;
+
+    Audio_SequencePlayerDisable(seqPlayer);
+    seqPlayer->unk_0b2 = 0;
+    seqPlayer->delay = 0;
+    seqPlayer->state = 1;
+    seqPlayer->fadeTimer = 0;
+    seqPlayer->fadeTimerUnkEu = 0;
+    seqPlayer->tempoAcc = 0;
+    seqPlayer->tempo = 120 * TATUMS_PER_BEAT; // 120 BPM
+    seqPlayer->unk_0C = 0;
+    seqPlayer->transposition = 0;
+    seqPlayer->noteAllocPolicy = 0;
+    seqPlayer->shortNoteVelocityTable = gDefaultShortNoteVelocityTable;
+    seqPlayer->shortNoteDurationTable = gDefaultShortNoteDurationTable;
+    seqPlayer->unk_E0 = 0;
+    seqPlayer->fadeVolume = 1.0f;
+    seqPlayer->fadeVelocity = 0.0f;
+    seqPlayer->volume = 0.0f;
+    seqPlayer->muteVolumeScale = 0.5f;
+
+    for (i = 0; i < 0x10; i++) {
+        Audio_SequenceChannelInit(seqPlayer->channels[i]);
+    }
+}
+
+void func_800EC734(s32 player) {
+    SequenceChannel *seqChannel;
+    SequencePlayer *seqPlayer;
```

Comment:
run the formatter on all files real quick, please and ty 
(itll catch star on the right and move it left)

Reply:
also.. this is very obviously a different context so it might not be an issue, but we use the word "player" to refer to link/the player everywhere in this codebase. do you think itd be good to make the separation?

Reply:
Formatting fixed. Confusion over "player" is a good point; I'll rename player -> seqPlayerIndex in one place. I think seqPlayer is fairly clear however, especially in context.

Reply:
we commoly use the abbreviation Idx throughout the code for index, if you wanted to shorten it. Its not a hard requirement, but it might look nice

Reply:
Surely, done for a bunch of function parameters. (But left the struct members.) Idx is not super commonly used, I'll note:
```
$ rg Index | wc -l
2200
$ rg Idx | wc -l
290
```

Reply:
maybe it just feels like that as of late lmao. Maybe something we want to standardize? @Roman971

Reply:
Never been a fan of "idx", mostly because it feels like a variant of "id", which can easily lead to confusion between an id and an index which are often vastly different. We should probably discuss it further tho.

Reply:
I like "idx" just to throw my hat into the pile (or whatever the phrase is), but I'm not against "index" either I guess